### DISCUSSION TITLE
I am trying to implement hover:text-color in group hover, it is not working.

### DISCUSSION THREAD

**Discussion by ProPhycient5** [tailwindlabs/tailwindcss#4808] [Help]
I am trying to bring group hover effect, most of the properties are working with group effect, but text-color effect is not working.
Please some idea or suggestion how to make it happen.
This is the className of that div that I am trying on.
`className="bg-black bg-opacity-10 pb-2 border-t border-b border-l border-black hover:text-white hover:bg-gray-400 hover:bg-opacity-20"`

Please where I am going wrong, what needs to be changed??????🤔🤔🤔🤔🤔🤔🤔
Below is the code of tailwind.config.js:
```
module.exports = {
  purge: ['./pages/**/*.{js,ts,jsx,tsx}', './components/**/*.{js,ts,jsx,tsx}'],
  darkMode: false, // or 'media' or 'class'
  theme: {
    extend: {
      colors: {
        primary: '#2680eb',
        'dar-gray': '#4b4b4b',
        'light-gray-0': '#eaeaea',
        'light-gray-1': 'rgb(75,75,75)',
        'light-gray-2': 'rgb(128,128,128)',
        'renderer-gray': 'rgb(224, 224, 224)',
        red: '#e34850',
        'green-400': '#2d9d78',
        'green-500': '#268e6c',
      },
      animation: {
        // If you are using Toast component
        left: 'left 0.3s',
        right: 'right 0.3s',
        // if you are using the animate variant of the modal
        modal: 'modal 0.3s',
        // if you are using drawer variant right
        'drawer-right': 'drawer-right 0.3s',
        // if you are using drawer variant left
        'drawer-left': 'drawer-left 0.3s',
        // if you are using drawer variant top
        'drawer-top': 'drawer-top 0.3s',
        // if you are using drawer variant bottom
        'drawer-bottom': 'drawer-bottom 0.3s',
      },
      keyframes: {
        // if you are using drawer variant right
        'drawer-right': {
          '0%, 100%': { right: '-500px' },
          '100%': { right: '0' },
        },
        // if you are using drawer variant left
        'drawer-left': {
          '0%, 100%': { left: '-500px' },
          '100%': { left: '0' },
        },
        // if you are using drawer variant top
        'drawer-top': {
          '0%, 100%': { top: '-500px' },
          '100%': { top: '0' },
        },
        // if you are using drawer variant bottom
        'drawer-bottom': {
          '0%, 100%': { bottom: '-500px' },
          '100%': { bottom: '0' },
        },
        // if you are using the animate variant of the modal
        modal: {
          '0%, 100%': { top: '-500px' },
          '100%': { top: '0' },
        },
        /* If you are using Toast component*/
        left: {
          '0%, 100%': { transform: 'translateX(-100%)' },
          '100%': { transform: 'translateX(0)' },
        },
        right: {
          '0%, 100%': { transform: 'translateX(100%)' },
          '100%': { transform: 'translateX(0)' },
        },
      },
    },
    variants: {
      /* If you are using Collapse or Accordion component*/
      transitionProperty: {
        height: 'height',
      },
    },
  },
  variants: {
    extend: {
      backgroundColor: ['active', 'checked'],
      inset: ['checked'],
      opacity: ['disabled'],
      textColor: ['active','visited', 'first', 'group-hover'],
      divideColor:['group-hover'],
    },
  },
  plugins: [],
}
```

Awaiting for your help.

**Comment by ProPhycient5** [comment]
O sorry, I don't know, I thought it would be required. So, if I keep one
variant, will it work  ?????

**Reply by smooth80** [reply]
good on ios whats the platform

**Comment by VicGUTT** [comment]
Hey I'm not sure I fully understand your use case. Is it that you want the element which has the class you provided to change it's text color to white, it's background color to gray-400 and it's opacity to 20 when the group is hovered ?

If so your template should look similar to:

```html
<div class="group">
    <p class="group-hover:text-white group-hover:bg-gray-400 group-hover:bg-opacity-20">abc</p>
</div>
```

In the example above there is a wrapper/parent element with the `group` class and elements inside of it should make use of the `group-*` variants.
Here's a quick demo : https://play.tailwindcss.com/embLabYQmn.

If you're not in JIT mode, make sure the `group-*` variants are set for the core plugins you're trying to apply it to.
The Tailwind config provided in your comment is missing the said `group-*` variants from the "backgroundColor" and "opacity" core pugins, so instead it should be:

```js
variants: {
    extend: {
      backgroundColor: ['active', 'checked', 'group-hover'],
      inset: ['checked'],
      opacity: ['disabled', 'group-hover'],
      textColor: ['active', 'visited', 'first', 'group-hover'],
      divideColor:['group-hover'],
    },
  },
```

Checkout the docs for more infos : https://tailwindcss.com/docs/hover-focus-and-other-states#group-hover

Hope it helps ✌🏾

**Reply by andreasvirkus** [reply]
Your play example works, but as soon as I set a name for the group, it stops working 🤔 

https://play.tailwindcss.com/hMkvNJyQwB

**Reply by wongjn** [reply]
Group names where [introduced in `v3.2`](https://tailwindcss.com/blog/tailwindcss-v3-2#nested-group-and-multiple-peer-support-using-variant-modifiers) – the Tailwind version in your Tailwind Play is set to `v2.2.19`.

**Comment by ProPhycient5** [comment]
I have tried your suggestion both in code and tailwind.config.js file. But
still it's not working.

I have attached the screenshot of the above changes.
Please check whether it is fine or not.

**Comment by ProPhycient5** [comment]
Background color and opacity are working, but not text-color, even
text-decoration like underline is working together, but not text-color. I
am not getting why is it not working.

**Reply by devntv** [reply]
> Background color and opacity are working, but not text-color, even text-decoration like underline is working together, but not text-color. I am not getting why is it not working. On Sun, 27 Jun 2021 at 15:17, Sawan Kumar Das ***@***.***> wrote:
> […](#)
> I have tried your suggestion both in code and tailwind.config.js file. But still it's not working. I have attached the screenshot of the above changes. Please check whether it is fine or not. On Sun, 27 Jun 2021 at 14:17, Victor Schedlyn GUTT < ***@***.***> wrote: > Hey I'm not sure I fully understand your use case. Is it that you want > the element which has the class you provided to change it's text color to > white, it's background color to gray-400 and it's opacity to 20 when the > group is hovered ? > > If so your template should look similar to: > > <div class="group"> > > <p class="group-hover:text-white group-hover:bg-gray-400 group-hover:bg-opacity-20">abc</p> > </div> > > In the example above there is a wrapper/parent element with the group > class and elements inside of it should make use of the group-* variants. > Here's a quick demo : https://play.tailwindcss.com/embLabYQmn. > > If you're not in JIT mode, make sure the group-* variants are set for > the core plugins you're trying to apply it to. > The Tailwind config provided in your comment is missing the said group-* > variants from the "backgroundColor" and "opacity" core pugins, so instead > it should be: > > variants: { > > extend: { > > backgroundColor: ['active', 'checked', 'group-hover'], > > inset: ['checked'], > > opacity: ['disabled', 'group-hover'], > > textColor: ['active', 'visited', 'first', 'group-hover'], > > divideColor:['group-hover'], > > }, > > }, > > Checkout the docs for more infos : > https://tailwindcss.com/docs/hover-focus-and-other-states#group-hover > > Hope it helps ✌🏾 > > — > You are receiving this because you authored the thread. > Reply to this email directly, view it on GitHub > <[#4808 (comment)](https://github.com/tailwindlabs/tailwindcss/discussions/4808#discussioncomment-926839)>, > or unsubscribe > <https://github.com/notifications/unsubscribe-auth/AQ6ETRMOXWR2TBSBTOTJ2WTTU3QSXANCNFSM47KMYJZA> > . >

try changing other colors to see if they work or not?. I tried your code and it works with textcolor

**Comment by ProPhycient5** [comment]
No, other text-colors are also not working on that group-hover, while it is
working on another div with *hover:text-white hover:underline* . I have
already sent the className of that div. I have followed all
your instructions, still I am not getting it.

**Reply by satz7** [reply]
Did you get any answer? i have also same problem
